src/index.ts

```diff
@@ -1,923 +1,1064 @@
+      const hha_hash = url.searchParams.get('hha_hash');
+      log.normal("%s (%s) connection for HHA ID: %s", anonymous ? "Anonymous" : "Agent", agent_id, hha_hash);
+
+      if (anonymous) {
+        log.debug("Adding Agent (%s) to anonymous list with HHA ID %s", agent_id, hha_hash);
+        this.anonymous_agents[agent_id] = hha_hash;
+      }
+
+      socket.on("close", async () => {
+        log.normal("Socket is closing for Agent (%s) using HHA ID %s", agent_id, hha_hash);
+
+        if (anonymous) {
+          log.debug("Remove anonymous Agent (%s) from anonymous list", agent_id);
+          delete this.anonymous_agents[agent_id];
+        }
+      });
+    });
 
-	await this.http_server.close();
-	log.info("HTTP server is closed");
+    this.ws_server.register("holo/wormhole/event", async ([agent_id]) => {
+      log.normal("Initializing wormhole setup for Agent (%s)", agent_id);
+      const event = `${agent_id}/wormhole/request`;
+
+      try {
+        log.debug("Registering RPC WebSocket event (%s) in namespace: %s", event, this.opts.NS);
+        this.ws_server.event(event, this.opts.NS);
+      } catch (e) {
+        if (e.message.includes('Already registered event'))
+          log.warn("RPC WebSocket event '%s' is already registered for Agent (%s)", event, agent_id);
+        else {
+          log.error("Failed during RPC WebSocket event registration: %s", String(e));
+          console.error(e);
+        }
+      }
+
+      return event;
+    }, this.opts.NS);
+
+    this.ws_server.register("holo/wormhole/response", async ([payload_id, signature]) => {
+      log.normal("Received signing response #%s with signature: %s", payload_id, signature);
+
+      // - match payload ID to payload
+      const [payload, f, r, toid] = this.pending_signatures[payload_id];
+
+      // - respond to HTTP request
+      f(signature);
+
+      // clear fallback timeout response
+      clearTimeout(toid);
+
+      // - return success
+      return true;
+    }, this.opts.NS);
+
+
+    // ------------------------------------------------------------------------
+
+    // Envoy - New Hosted Agent Sign-up Sequence
+    this.ws_server.register("holo/agent/signup", async ([hha_hash, agent_id]) => {
+      log.normal("Received sign-up request from Agent (%s) for HHA ID: %s", agent_id, hha_hash)
+
+      const failure_response = (new HoloError("Failed to create a new hosted agent")).toJSON();
+      let resp;
+
+      log.info("Retrieve the holo-hosting-app cell id using the Installed App Id: '%s'", HHA_INSTALLED_APP_ID);
+      const appInfo = await this.callConductor("internal", { installed_app_id: HHA_INSTALLED_APP_ID });
+
+      if (!appInfo) {
+        log.error("Failed during HHA AppInfo lookup: %s", appInfo);
+        return failure_response;
+      }
+
+      const hha_cell_id = appInfo.cell_data[0][0];
+      const buffer_host_agent_hha_id = hha_cell_id[1];
+
+      log.info("Look-up Hosted App's HHA record using ID '%s'", hha_hash);
```

Comment:
I don't see the need for this lookup. 

There are multiple checks before you get here. 
1. the fact that the KV store routed the user here is proof that its the right hha_id.
After this point:
1. It will fail while installing an instance for this agent.


Also, It would make more sense to check if happ is activated by the host.

Reply:
The upside of removing zome call is a faster sign-up

Reply:
This isn't to prove the hha_id is correct, but to retrieve the dnas associated with this happ, in order to install them in the conductor.

Reply:
Ok, we can remove it when we do our version update since we won't need the DNA URL at that point